Amazon.IonObjectMapper.Test/IonObjectSerializerTest.cs

```diff
@@ -619,5 +619,77 @@ private void AssertIsTruck(object actual)
         {
             Assert.AreEqual(actual.ToString(), TestObjects.nativeTruck.ToString());
         }
+
+        [TestMethod]
+        public void TestSerializingAndDeserializingPropertiesWithAccessModifiers()
+        {
+            var classWithProperties = new ClassWithProperties(PublicProperty: "PublicProperty", DefaultProperty: "DefaultProperty",
+                ProtectedProperty: "ProtectedProperty", ProtectedInternalProperty: "ProtectedInternalProperty", InternalProperty: "InternalProperty",
+                PrivateProperty: "PrivateProperty", PrivateProtectedProperty: "PrivateProtectedProperty");
+
+            var deserializedObject = Serde(classWithProperties);
+
+            Assert.AreEqual("PublicProperty", deserializedObject.PublicProperty);
+            Assert.AreEqual("", deserializedObject.GetDefaultProperty());
+            Assert.AreEqual("", deserializedObject.GetProtectedProperty());
+            Assert.AreEqual("ProtectedInternalProperty", deserializedObject.ProtectedInternalProperty);
+            Assert.AreEqual("InternalProperty", deserializedObject.InternalProperty);
+            Assert.AreEqual("", deserializedObject.GetPrivateProperty());
+            Assert.AreEqual("", deserializedObject.GetPrivateProtectedProperty());
+        }
+
+        [TestMethod]
+        public void TestSerializingAndDeserializingReadOnlyPropertiesWithAccessModifiers()
+        {
+            var classWithReadonlyProperties = new ClassWithReadonlyProperties(PublicProperty: "PublicProperty", DefaultProperty: "DefaultProperty",
+                ProtectedProperty: "ProtectedProperty", ProtectedInternalProperty: "ProtectedInternalProperty", InternalProperty: "InternalProperty",
+                PrivateProperty: "PrivateProperty", PrivateProtectedProperty: "PrivateProtectedProperty");
+
+            var deserializedObject = Serde(classWithReadonlyProperties);
+
+            Assert.AreEqual("PublicProperty", deserializedObject.PublicProperty);
```

Comment:
Looks like a fair bit of repetition between the tests. Let's define a utility function to handle this.

Reply:
Reduced code duplication by creating `Check` function which takes in expected string and asserts against the object's string representation.